ui/src/app/components/gke-app-configuration-panels/create-gke-app.tsx

```diff
@@ -150,6 +187,81 @@ export const CreateGkeApp = ({
           {...{ appType, machine, persistentDiskRequest }}
         />
       </div>
+      <div style={{ ...styles.controlSection }}>
+        <FlexRow
+          style={{
+            alignItems: 'center',
+          }}
+        >
+          <CheckBox
+            aria-label={`gke-autodelete-checkbox`}
+            disabled={isAppActive(app)}
+            checked={
+              createAppRequest.autodeleteEnabled ||
+              (app && app.autodeleteEnabled)
+            }
+            onChange={(autodeleteEnabled) => {
+              setCreateAppRequest((prevState) => ({
+                ...prevState,
+                autodeleteEnabled,
+              }));
+              setAutodeleteDropdownChecked(autodeleteEnabled);
+            }}
+            style={{ marginRight: '1rem', zoom: 1.5 }}
+          />
+          <FlexColumn>
+            <label style={styles.label} htmlFor='gke-autodelete-label'>
+              Automatically delete application after
+            </label>
+            <p style={{ marginTop: '0' }}>
+              Your persistent disk will not be deleted.
+            </p>
+          </FlexColumn>
+          <TooltipTrigger side='top' content={autodeleteToolTip}>
+            <div>
+              <ClrIcon
+                className='is-solid'
+                style={{
+                  marginRight: '0.5rem',
+                  marginTop: '0.3rem',
+                  zoom: 1.2,
+                }}
+                shape='info-standard'
+              />
+            </div>
+          </TooltipTrigger>
+          <FlexColumn>
+            <Dropdown
+              aria-label={`gke-autodelete-dropdown`}
+              appendTo='self'
+              disabled={isAppActive(app) || !autodeleteDropdownChecked}
+              options={Array.from(AutodeleteMinuteThresholds.entries()).map(
```

Comment:
This conversion is awkward.  We can use a more appropriate data structure here.  How about this:

```
autodeleteDayThresholds = [1,3,7,8,15,30]
autodeleteDayThresholds.map(days => ({
  value: days * 24 * 60, 
  label: `Idle for ${days} days`
})
```

Reply:
ah okay. I copied this conversion from autopause